packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -1024,18 +1137,28 @@ export default class GoogleCalendarService implements Calendar {
 
     // Populate the cache back for the remaining calendars, if any
     const remainingCalendars =
-      calendarsWithSameCredentialId.filter((sc) => sc.externalId !== calendarId) || [];
+      calendarsWithSameCredential.filter(
+        (sc) => sc.externalId !== calendarId && sc.integration === this.integrationName
```

Comment:
Better to check integration as well like we do at other places.

integration(google_calendar) + externalId(abc@gmail.com) together uniquely identifies a calendar